wandb/apis/workspaces/internal.py

```diff
@@ -0,0 +1,279 @@
+        validate_assignment=True,
+        populate_by_name=True,
+        arbitrary_types_allowed=True,
+    )
+
+
+class OuterSectionSettings(WorkspaceAPIBaseModel):
+    smoothing_weight: Annotated[float, Ge(0)] = 0
+    smoothing_type: str = "exponential"
+    x_axis: str = "_step"
+    ignore_outliers: bool = False
+    use_runs_table_grouping_in_panels: bool = True
+    x_axis_min: Optional[Annotated[float, Ge(0)]] = None
+    x_axis_max: Optional[Annotated[float, Ge(0)]] = None
+    color_run_names: Optional[bool] = None
+    max_runs: Optional[int] = None
+    point_visualization_method: Optional[Literal["bucketing-gorilla", "sampling"]] = (
+        None
+    )
+    suppress_legends: Optional[bool] = None
+    tooltip_number_of_runs: Optional[Literal["single", "default", "all_runs"]] = None
+
+    @computed_field
+    @property
+    def smoothing_active(self) -> bool:
+        return self.smoothing_type != "none" or self.smoothing_weight != 0
+
+    @computed_field
+    @property
+    def x_axis_active(self) -> bool:
+        return (
+            self.x_axis != "_step"
+            or self.x_axis_min is not None
+            or self.x_axis_max is not None
+        )
+
+
+class OuterSection(WorkspaceAPIBaseModel):
+    panel_bank_config: PanelBankConfig
+    panel_bank_section_config: PanelBankSectionConfig
+
+    # this is intentionally dict because it has arbitrary keys (the run ids)
+    custom_run_colors: dict
+
+    name: str = ""
+    run_sets: LList[Runset] = Field(default_factory=lambda: [Runset()])
+    ref: Ref = Field(default_factory=Ref)
+    settings: OuterSectionSettings = Field(default_factory=OuterSectionSettings)
+    open_run_set: int = 0
+    open_viz: bool = True
+
+
+# unfortunate nomenclature... this is actually a workspace's view's spec...
+class WorkspaceViewspec(WorkspaceAPIBaseModel):
+    section: OuterSection
+    viz_expanded: bool = False
+    library_expanded: bool = True
+    ref: Ref = Field(default_factory=Ref)
+
+
+class View(WorkspaceAPIBaseModel):
+    entity: str
+    project: str
+    display_name: str
+    name: str
+    id: str
+    viewspec: WorkspaceViewspec
+
+
+def upsert_view2(view: View, clone: bool = False) -> Dict[str, Any]:
+    query = gql(
+        """
+        mutation UpsertView2($id: ID, $entityName: String, $projectName: String, $type: String, $name: String, $displayName: String, $description: String, $spec: String, $parentId: ID, $locked: Boolean, $previewUrl: String, $coverUrl: String, $showcasedAt: DateTime, $createdUsing: ViewSource) {
+        upsertView(
+            input: {id: $id, entityName: $entityName, projectName: $projectName, name: $name, displayName: $displayName, description: $description, type: $type, spec: $spec, parentId: $parentId, locked: $locked, previewUrl: $previewUrl, coverUrl: $coverUrl, showcasedAt: $showcasedAt, createdUsing: $createdUsing}
+        ) {
+            view {
+            id
+            ...ViewFragmentMetadata2
```

Comment:
Do we really need to return all these fields? It seems like only view.name is used `resp["upsertView"]["view"]["name"]`

Reply:
Agreed.  Can you help me narrow down just the fields that are necessary?  When I tried just uploading a handful of fields like name and spec, the page showed an infinite spinner, so I probably deleted too much...